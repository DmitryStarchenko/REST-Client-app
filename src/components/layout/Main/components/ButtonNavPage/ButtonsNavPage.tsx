import { useTranslations } from 'next-intl';
import React from 'react';

import { Link } from '@/i18n/navigation';
import { ReadonlyFC } from '@/types/readonly.types';

import styles from './ButtonsNavPage.module.css';

const ButtonsNavPage: ReadonlyFC = () => {
  const translationMain = useTranslations('Main');

  return (
    <>
      <Link className={styles.navButton} href={'/client'}>
        {translationMain('client')}
      </Link>
      <Link className={styles.navButton} href={'/variables'}>
        {translationMain('variables')}
      </Link>
      <Link className={styles.navButton} href={'/history'}>
        {translationMain('history')}
      </Link>
    </>
  );
};

export default ButtonsNavPage;
