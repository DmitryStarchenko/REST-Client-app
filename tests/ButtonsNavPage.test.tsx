import { render, screen } from '@testing-library/react';
import { useTranslations } from 'next-intl';
import React from 'react';
import { beforeEach, describe, expect, it, Mock, vi } from 'vitest';

import ButtonsNavPage from '../src/components/layout/Main/components/ButtonNavPage/ButtonsNavPage';

vi.mock('next-intl', () => ({
  useTranslations: vi.fn(),
}));

vi.mock('next/link', () => ({
  default: vi.fn(({ children, href, className }) => (
    <a href={href} className={className}>
      {children}
    </a>
  )),
}));

vi.mock('@/i18n/navigation', () => ({
  Link: vi.fn(({ children, href, className }) => (
    <a href={href} className={className}>
      {children}
    </a>
  )),
}));

vi.mock('./ButtonsNavPage.module.css', () => ({
  default: {
    navButton: 'navButton-class',
  },
}));

describe('ButtonsNavPage', () => {
  const mockTranslation = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();
    mockTranslation.mockImplementation((key: string) => {
      const translations: Record<string, string> = {
        client: 'Client Page',
        variables: 'Variables Page',
        history: 'History Page',
      };
      return translations[key] || key;
    });

    (useTranslations as Mock).mockReturnValue(mockTranslation);
  });

  it('should render all navigation buttons with correct text and links', () => {
    render(<ButtonsNavPage />);
    const clientLink = screen.getByRole('link', { name: 'Client Page' });
    const variablesLink = screen.getByRole('link', { name: 'Variables Page' });
    const historyLink = screen.getByRole('link', { name: 'History Page' });

    expect(clientLink).toBeInTheDocument();
    expect(clientLink).toHaveAttribute('href', '/client');
    expect(variablesLink).toBeInTheDocument();
    expect(variablesLink).toHaveAttribute('href', '/variables');
    expect(historyLink).toBeInTheDocument();
    expect(historyLink).toHaveAttribute('href', '/history');
  });

  it('should call useTranslations with correct namespace', () => {
    render(<ButtonsNavPage />);
    expect(useTranslations).toHaveBeenCalledWith('Main');
  });

  it('should call translation function with correct keys', () => {
    render(<ButtonsNavPage />);
    expect(mockTranslation).toHaveBeenCalledWith('client');
    expect(mockTranslation).toHaveBeenCalledWith('variables');
    expect(mockTranslation).toHaveBeenCalledWith('history');
    expect(mockTranslation).toHaveBeenCalledTimes(3);
  });

  // it('should match snapshot', () => {
  //   const { container } = render(<ButtonsNavPage />);
  //   expect(container).toMatchSnapshot();
  // });
});
