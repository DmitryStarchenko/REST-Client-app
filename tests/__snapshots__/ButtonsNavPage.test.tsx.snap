// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonsNavPage > should match snapshot 1`] = `
<div>
  <a
    class="_navButton_d0c136"
    href="/client"
  >
    Client Page
  </a>
  <a
    class="_navButton_d0c136"
    href="/variables"
  >
    Variables Page
  </a>
  <a
    class="_navButton_d0c136"
    href="/history"
  >
    History Page
  </a>
</div>
`;
